pmd-core/src/main/java/net/sourceforge/pmd/cache/CachedRuleViolation.java

```diff
@@ -14,7 +14,7 @@
 /**
  * A {@link RuleViolation} implementation that is immutable, and therefore cache friendly
  */
-public final class CachedRuleViolation implements RuleViolation {
+public class CachedRuleViolation implements RuleViolation {
```

Comment:
if you are not dealing with cache now, please revert these changes. On their own they make little sense